src/optimizer.cpp

```diff
@@ -76,11 +77,27 @@ geometry_msgs::msg::TwistStamped Optimizer::evalControl(
 {
   for (size_t i = 0; i < iteration_count_; ++i) {
     generated_trajectories_ = generateNoisedTrajectories(robot_pose, robot_speed);
-    auto && costs = critic_manager_.evalTrajectoriesScores(generated_trajectories_, plan, robot_pose);
+    auto && costs =
+      critic_manager_.evalTrajectoriesScores(generated_trajectories_, plan, robot_pose);
     updateControlSequence(costs);
   }
 
-  return getControlFromSequenceAsTwist(0, plan.header.stamp);
+  auto control = getControlFromSequenceAsTwist(0, plan.header.stamp);
+
+  // TODO calc shift from controller freq
+  shiftControlSequence(1);
```

Comment:
What's with this?

Reply:
It is feature that shift control sequence by one time step, after every control made. It could be useful for next iteration

Reply:
Got it -- can the TODO be removed then or is there more to be done here?

Reply:
It would work now only if controller period = model_dt = 0.1s now. Probably need to evaluate this, or have option to set this value. I think just add parameter

Reply:
Added parameter

Reply:
If it can be derived from the existing information, we should do that versus having 2 parameters that are critically tied by a function that might not be clear to a user

Reply:
I find that people tend to randomly change numbers in parameters they dont understand

Reply:
Then we need to take in controller frequency

Reply:
You can do that pretty easily https://github.com/ros-planning/navigation2/blob/main/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L149

basically, just don't add the `name` namespace to the parameter. The plugins have access to the server-level parameters

Reply:
implement logic for evaluating if we can shift